### DISCUSSION TITLE
pb admin dashboard and collection with heavy fields

### DISCUSSION THREAD

**Discussion by yureckey** [pocketbase/pocketbase#3974] [Q&A]
I'm running some queries with 1mb (or 2 or 3) json field to save/read. I noted that in admin dashboard even when I have toggled column with big json off - all data is still loaded (which take time). I guess, toggle columns isn't used in fields query parameters. I think it would be good to be able to disable loading of certain fields in view of pb admin collections

**Comment by ganigeorgiev** [comment]
Yes, currently the "Toggle columns" control only toggles the listing column visibility on the client-side.

I'll have to think a little more on this as adding the toggled columns to the `fields` query parameter will require refetching the list results again with every change which could be even slower.
Eventually we may add a custom `fields` modifier similar to the `:excerpt(100)` (_or maybe change `excerpt` to work also with json fields and to return a string based version?_).

With that said, note that storing very large column text values in SQLite may slow significantly your queries (especially when combined with multiple `JOIN` statements or aggregate functions) and it may be better such large `json` data to be stored as `file` field.

**Reply by yureckey** [reply]
Got it, thanks!

**Comment by jonshipman** [comment]
You could also go EAV and have a `meta` table with a relation back to the parent table. For big datasets, its usually preferable to have the bare minimum (basically sort columns) on the main table and data in another. Just remember to enable the cascade deletion.

**Reply by yureckey** [reply]
Got it, thanks!
